src/components/checkbox/Checkbox.tsx

```diff
@@ -55,7 +55,7 @@ const Checkbox: GenericComponent<CheckboxProps> = forwardRef<HTMLInputElement, C
     const IconCheckmark = size === 'large' ? IconCheckmarkMediumOutline : IconCheckmarkSmallOutline;
 
     const restProps = omit(others, ['onChange']);
-    const boxProps = pickBoxProps(restProps);
+    const boxProps = { ...pickBoxProps(restProps), onClick: restProps.onClick };
```

Comment:
onClick will still also be in the inputProps. But I guess we can ignore that, since the input isn't actually visible/clickable.